node/src/actors/chain_manager/mod.rs

```diff
@@ -1356,6 +1356,19 @@ impl ChainManager {
         Box::new(fut)
     }
 
+    /// Transition the ChainManager state machine into a new state.
+    ///
+    /// This is expected to be the only means for updating the state machine, so debugging is easier
+    /// and to ensure that every transition gets logged in a predictable format.
+    fn update_state_machine(&mut self, new_state: StateMachine) {
+        log::debug!(
```

Comment:
Should we check if `new_state == self.sm_state` and print a different log in that case?

Reply:
Done!